File: Code-CSharp/Base/String/StringViewEntry.cs
namespace CodeCsharp.Base.String; 

public class StringViewEntry {
  public static void StringViewProgram() {
    //initialize string
    string s = string.Empty;
    //initialize string view
    ReadOnlySpan<char> view = s.AsSpan();
    //get char from view
    char c = view[1];
    //length of view
    int len = view.Length;
    //whether view empty
    bool b1 = view.IsEmpty;
    //sub string view
    ReadOnlySpan<char> sub = view.Slice(1,2);
    //compare string view
    int result = view.CompareTo(sub,StringComparison.Ordinal);
    //build new string from view
    string s1 = view.ToString();
  }
  public static bool TestAll() {
    bool ret = true;
    string s = "Example";
    ReadOnlySpan<char> view = s.AsSpan();
    ret = view[1] == 'x'
          && view.Length == 7
          && !view.IsEmpty;
    ReadOnlySpan<char> sub = view.Slice(1, 2);
    ret = ret
          && sub.ToString().Equals("xa")
    && view.CompareTo(sub,StringComparison.Ordinal) < 0;
    string s2 = string.Empty;
    view = s2.AsSpan();
    ret = ret && view.IsEmpty;
    if (!ret)
      Console.WriteLine("String View Test Failed!");
    return ret;
  }
}

File: Code-Cpp/code_cpp/base/string/string_view_entry.cc
#include "string_view_entry.h"
#include <iostream>
void StringViewEntry::StringViewProgram() {
  //initialize string
  std::string s = std::string("Example");
  //initialize string view
  std::string_view view = s;
  //get char from view
  char c = view[1];
  //length of view
  int size = view.length();
  //whether view empty
  bool b1 = view.empty();
  //sub string view
  std::string_view sub = view.substr(1, 2);
  //compare string view
  int result = view.compare(sub);
  //build new string from view
  std::string s1 = std::string(view);
}
bool StringViewEntry::TestAll() {
  bool ret = true;
  using namespace std;
  string s = "Example";
  string_view view = s;
  ret = view[1] == 'x'
      && view.length() == 7
      && !view.empty();
  string_view sub = view.substr(1, 2);
  ret = ret
      && sub == "xa"
      && view.compare(sub) <0;
  view = string_view(s.end(), s.end());
  ret = ret && view.empty();
  if (!ret)
    cout << "String View Test Failed!";
  return ret;
}
